pages/index.tsx

```diff
@@ -8,10 +8,60 @@ import {
   Value,
   Table,
 } from "../components";
-import UnstyledInfoIcon from "../public/icons/info.svg";
-import { formatMillions, QUERIES } from "../utils";
 
-const IndexPage: React.FC = () => {
+import {
+  getContenfulClient,
+  ContentfulSynth,
+  formatMillions,
+  QUERIES,
+  errorFilter,
+} from "../utils";
+import { GetServerSideProps, InferGetServerSidePropsType } from "next";
+import { client, Emp } from "../utils/umaApi";
+import { ethers } from "ethers";
+
+export const getServerSideProps: GetServerSideProps = async () => {
+  const contentfulClient = getContenfulClient();
+  const {
+    items: cmsItems,
+  } = await contentfulClient.getEntries<ContentfulSynth>({
+    content_type: "synth",
+  });
+  const cmsData = cmsItems.map((item) => item.fields);
+
+  const data: Emp[] = (
+    await Promise.all(
+      cmsData.map(async (synth) => {
+        try {
+          const stats = await client.getEmpStats(synth.address);
+          const state = await client.getEmpState(synth.address);
+          return {
+            ...stats,
+            ...state,
+            isActive: Date.now() < Number(state.expirationTimestamp) * 1000,
+            ...synth,
+          };
+        } catch (err) {
+          // may be worth to make an error class for this specific thing
+          return new Error(JSON.stringify(synth));
+        }
+      })
+    )
+  ).filter(errorFilter) as Emp[];
+
```

Comment:
This function seems to be the same as the one in [address].tsx, might be worth making a helper function.